gaia-header.js

```diff
@@ -364,11 +371,51 @@ module.exports = component.register('gaia-header', {
   observerStop: function() {
     if (!this.observing) { return; }
     this.observer.disconnect();
+
     this.observing = false;
     debug('observer stopped');
   },
 
   /**
+   * Handle DOM events.
+   * @param {Event} The DOM Event that's being handled.
+   *
+   * @private
+   */
+  handleEvent: function(e) {
+    debug('received event', e.type, e);
+    switch(e.type) {
+      case 'resize':
+        this.onResize();
+        break;
+    }
+  },
+
+  /**
+   * Handle 'resize' events.
+   *
+   * @private
+   */
+  onResize: function() {
+    debug('onResize', this._resizeThrottlingId);
+
+    if (this._resizeThrottlingId !== null) {
+      return;
+    }
+
+    if (this.notFlush) {
```

Comment:
I'm not sure if this should be here. Just because the header isn't flush with the window edge doesn't mean that it won't require `runFontFit()` when the window is resized. For example I may have a header that spans 2/3 of the window width instead of 100%.